src/Transformer/EnumTransformerFactory.php

```diff
@@ -20,7 +20,7 @@ protected function createTransformer(Type $sourceType, Type $targetType, MapperM
         }
 
         // target is enum, source isn't
-        if (!$this->isEnumType($sourceType) && $this->isEnumType($targetType, true)) {
+        if (!$this->isEnumType($sourceType) && $this->isEnumType($targetType, true) && null !== $targetType->getClassName()) {
```

Comment:
```suggestion
        if (!$this->isEnumType($sourceType) && $this->isEnumType($targetType, true)) {
```

and let's add
```php
/**
 * @phpstan-assert-if-true !null $type->getClassName()
 */
```
to `isEnumType()`